lib/presentation/widget/side_menu/side_menu_viewmodel.dart

```diff
@@ -29,15 +29,47 @@
 //  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
 //  the Additional Terms applicable to LinShare software.
 
+import 'package:dartz/dartz.dart';
+import 'package:domain/domain.dart';
+import 'package:linshare_flutter_app/presentation/redux/actions/authentication_action.dart';
 import 'package:linshare_flutter_app/presentation/redux/states/app_state.dart';
+import 'package:linshare_flutter_app/presentation/util/router/app_navigation.dart';
+import 'package:linshare_flutter_app/presentation/util/router/route_paths.dart';
 import 'package:linshare_flutter_app/presentation/widget/base/base_viewmodel.dart';
 import 'package:redux/src/store.dart';
+import 'package:redux_thunk/redux_thunk.dart';
 
 class SideMenuDrawerViewModel extends BaseViewModel {
+  final DeletePermanentTokenInteractor deletePermanentTokenInteractor;
+  final AppNavigation _appNavigation;
 
-  SideMenuDrawerViewModel(Store<AppState> store) : super(store);
+  SideMenuDrawerViewModel(
+    Store<AppState> store,
+    this._appNavigation,
+    this.deletePermanentTokenInteractor
+  ) : super(store);
 
   void goToMySpace() {
     // push via nested navigation here
   }
+
+  void logout() {
+    store.dispatch(logoutAction());
+  }
+
+  ThunkAction<AppState> logoutAction() {
+    return (Store<AppState> store) async {
+      await deletePermanentTokenInteractor.execute().then(
+          (result) => result.fold(
+              (failure) => store.dispatch(AuthenticationAction(Left(failure))),
```

Comment:
In case we can not log out success from server, I think it is better behavior that client side still can log out no matter what.

Reply:
agreed